releases.md

```diff
@@ -0,0 +1,51 @@
+# Release Process
+
+This process is a work-in-progress until the first major release of the project.
+
+The process will utimately aim to have:
+
+- A published release date for the first caikit major release, upon the availability of the first beta release
+- A well-known cyclical release dates for caikit minor and patch releases, upon the availability of the first caikit major release
+- Minor releases backwards compatible with respect to the API, upon the availability of the first caikit major release
+
+## Semantic versioning
+
+caikit versions are expressed as x.y.z, where x is the major version, y is the minor version, and z is the patch version, following [Semantic Versioning](https://semver.org/spec/v2.0.0.html) terminology.
+
+## Patch releases
+
+Patch releases provide users with bug fixes and security fixes. They do not contain new features.
+
+Patch releases should be done every second week on a Wednesday (assuming there are changes since the last release). A patch release to fix a high priority regression or security issue does not have to follow this schedule, but it is highly desirable that it is released on a Wednesday if possible.
```

Comment:
Do we do patch releases for the latest `major.minor` release only or backport as well?
If backport, how many releases do we support and which ones?

Reply:
From my experience, back porting becomes quite challenging to do. As a consequence, some communities  that I contributed to decided to backport/patch older releases only on the latest minor release.  Therefore, not to support backport. I think this works quite well once stated up front. What do you think @gabe-l-hart @afrittoli?

Reply:
Patching the latest minor release seems like a fair approach, especially if stated up front like mentioned by @hickeyma. 
Once the project is more mature/adopted I think it becomes important to make sure that users have a path to a security patch that does not involve going through backwards-incompatible changes unless the version in use is marked as end-of-life (EOL).